### PR TITLE
Feature/chesapeake cvpr training

### PR DESCRIPTION
**Pull request by calebrob6**
Small set of changes to improve the Chesapeake CVPR trainer:
- Nodata is always 0
- Patches that are too small are replaced with nodata
- Option to train on two class sets
- Option to use FCN model
- Option to use focal loss
- Val sampler uses GridGeoSampler

### CODE REVIEW

File: torchgeo/trainers/chesapeake.py [hunk 1]
```diff
@@ -31,8 +32,20 @@
 DataLoader.__module__ = "torch.utils.data"
 Module.__module__ = "torch.nn"
 
-CMAP = matplotlib.colors.ListedColormap(
-    [np.array(Chesapeake7.cmap[i + 1]) / 255.0 for i in range(6)]
+CMAP_7 = matplotlib.colors.ListedColormap(
```
Comment (adamjstewart) [inline torchgeo/trainers/chesapeake.py:35]: Can you add a TODO saying that this stuff needs to be moved to the `ChesapeakeCVPR` class in the future?

File: torchgeo/trainers/chesapeake.py [hunk 2]
```diff
@@ -45,31 +58,46 @@ class ChesapeakeCVPRSegmentationTask(LightningModule):
...
     def config_task(self) -> None:
         """Configures the task based on kwargs parameters passed to the constructor."""
+        if self.hparams["class_set"] == 5:
+            num_classes = 5
+            classes = [1, 2, 3, 4]
+        elif self.hparams["class_set"] == 7:
+            num_classes = 7
+            classes = [1, 2, 3, 4, 5, 6]
+        else:
+            raise ValueError("'class_set' must be either 5 or 7")
```
Comment (adamjstewart) [inline torchgeo/trainers/chesapeake.py:68]: This could be replaced with:
```python
num_classes = self.hparams["class_set"]
assert num_classes in [5, 7]
classes = list(range(num_classes))
```

**Comment by adamjstewart** [comment]
You can ignore the failing codecov checks for now, I'll just force-merge this when it's ready. For new contributors, and once the paper is finished, we'll start enforcing that all code is tested.

**Comment by calebrob6** [comment]
@adamjstewart is that all you wanted for this one?

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Feel free to force-merge or ping me once the remaining tests pass.
